File: src/Password.java
public class Password {
    private String password;

    public Password(String password) {
        this.password = password;
    }
    
    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public boolean compareCurrentPassword(String enteredPassword){
        return enteredPassword.equals(password);
    }

    public boolean checkPasswordCond(String newPassword){
		boolean containsUpperCase = checkUpperCaseCond(newPassword),
        containsLowerCase =checkUpperCaseCond(newPassword),
	    containsNumber = checkUpperCaseCond(newPassword),
        continsSpeacialCharacter=isContains(newPassword),
        lenght=lenghtCond(newPassword),
        valid=false;  
		if (lenght && continsSpeacialCharacter && containsUpperCase && containsLowerCase && containsNumber){
				valid=true;
			}
         return valid;
	}

    public boolean isContains(String newPassword){
        boolean valid=false;
        if(newPassword.contains(".") || newPassword.contains("_")){
            valid=true;
        }
        return valid;
    }

    public boolean lenghtCond(String newPassword){
        boolean valid=false;
        if(newPassword.length() > 8 && newPassword.length() < 20){
            valid=true;
        }
        return valid;
    }

    public boolean checkUpperCaseCond(String newPassword){
        boolean valid=false;
        for (char c : newPassword.toCharArray()) {
            if (Character.isUpperCase(c)) {
                valid = true;
                break;
            }
        }
        return valid;
    }

    public boolean checkLowerCaseCond(String newPassword){
        boolean valid=false;
         for (char c : newPassword.toCharArray()) {
            if (Character.isLowerCase(c)) {
                valid = true;
                break;
            }
        }
        return valid;
    }

    public boolean checkDigitCaseCond(String newPassword){
        boolean valid=false;
         for (char c : newPassword.toCharArray()) {
            if (Character.isDigit(c)) {
                valid = true;
                break;
            }
        }
        return valid;
    }
    
}

File: python_development/PersonObject/Password.py
class Password:
    def __init__(self, password):
        self.password = password

    def getPassword(self):
        return self.password

    def setPassword(self, password):
        self.password = password

    def compareCurrentPassword(self, enteredPassword):
        return enteredPassword == self.password

    def checkPasswordCond(self, newPassword):
        contains_upper_case = self.checkUpperCaseCond(newPassword)
        contains_lower_case = self.checkLowerCaseCond(newPassword)
        contains_number = self.checkNumberCaseCond(newPassword)
        contains_special_character = self.isContainsSpecialChar(newPassword)
        length_condition = self.lengthCond(newPassword)

        if length_condition and contains_special_character and contains_upper_case and contains_lower_case and contains_number:
            return True
        return False

    def isContainsSpecialChar(self, newPassword):
        if "." in newPassword or "_" in newPassword:
            return True
        else:
            print("The password must contain some special character such as '_' , '.'.")
            return False

    def lengthCond(self, newPassword):
        if 8 < len(newPassword) < 20:
            return True
        else:
            print("The password must be at least 8 and at most 20 characters.")
            return False

    def checkUpperCaseCond(self, newPassword):
        if any(char.isupper() for char in newPassword):
            return True
        print("The password must contain at least one uppercase letter.")
        return False

    def checkLowerCaseCond(self, newPassword):
        if any(char.islower() for char in newPassword):
            return True
        print("The password must contain at least one lowercase letter.")
        return False

    def checkNumberCaseCond(self, newPassword):
        if any(char.isdigit() for char in newPassword):
            return True
        print("The password must contain at least one number.")
        return False

    def equals(self, obj):
        if isinstance(obj, Password):
            return obj.getPassword() == self.getPassword()
        return False